### DISCUSSION TITLE
Does V have scanf?

### DISCUSSION THREAD

**Discussion by deemp** [vlang/v#14086] [General]
Assume in `stdin`, there are `3 4 abra`. I'd like to read a `3`, stop there and have ` 4 abra` still in the input. How can I do that easy way? 
In C++, we have `int a; cin >> a;` How can we emulate something similarly convenient in V? Maybe there is some function that I missed that reads until the next non-space character?

Context: I'd like to read some heterogeneous arguments from `stdin`, like in the example above. However, `os.input('')` reads [the whole line](https://modules.vlang.io/os.html#input). 
Okay, I can `os.input('').split(' ').map(it.int())`, but what if there is a `string` in the stdin? 

Well, I can make a sum type `type Mixer = int | string` and a map `mut args : map[int]Mixer` and then insert the arguments parsed with `atoi()` (and strings if it fails) there and access them by index. But how to get value of an `int` from a `Mixer` to further assign it to a variable without pattern matching? Simple casting isn't quite safe and seems to be impossible. 

So I think that reading a string until the next space character + `atoi()` with handling wrong parsing result would do. Just didn't want to implement it and decided to ask.

**Comment by changrui** [comment]
No, you can use `os.input(prompt string) string`.

**Reply by JalonSolov** [reply]
... and then convert with the simple methods such as `.int()` or `.f32()`, or use the `strconv` module routines for more complete conversion types, error checking, etc.

**Comment by medvednikov** [comment]
`os.input('enter number').f64()`

**Comment by deemp** [comment]
Sorry I updated the post to clarify my case. Thank you all!

**Reply by JalonSolov** [reply]
Now none of the previous replies make sense... a different discussion may have been better.

In any case, you're almost there.  Read the line with `os.input`, then split on spaces.  I would use an array of `string` values rather than a map (and that's what `.split` gives you anyway), since that will keep the pieces in order without having to have `int` keys.

Then you can use a `for` loop, with `strconv.atoi()` and `strconv.atof()` to attempt converting.  If both of them fail, you have a string.

**Comment by medvednikov** [comment]
V will have what you want, similar to C++/Java/C#, it's just not implemented yet.

**Reply by deemp** [reply]
Thank you for reply! Sounds :fire:
